sources/advanced/Command.ts

```diff
@@ -260,31 +270,121 @@ export abstract class Command<Context extends BaseContext = BaseContext> {
+
+    static Array(descriptor?: string, {hidden = false}: {hidden?: boolean} = {}) {
+        return <Context extends BaseContext>(prototype: Command<Context>, propertyName: string) => {
+            // The legacy style decorator.
+            // TODO: Remove in next major.
+            if (typeof descriptor === 'string') {
+                const optNames = descriptor.split(`,`);
+
+                this.registerDefinition(prototype, command => {
+                    command.addOption({names: optNames, arity: 1, hidden});
+                });
+
+                this.registerTransformer(prototype, (state, command) => {
+                    for (const {name, value} of state.options) {
+                        if (optNames.includes(name)) {
+                            // @ts-ignore: The property is meant to have been defined by the child class
+                            command[propertyName] = command[propertyName] || [];
+                            // @ts-ignore: The property is meant to have been defined by the child class
+                            command[propertyName].push(value);
+                        }
+                    }
+                });
+            } else {
+                const SPECIAL_KEY = `__command_array`;
+
+                this.registerTransformer(prototype, (state, command) => {
+                    // We create a temporary value store
+                    const values: unknown[] = [];
+
+                    // It's fine if we don't cleanup the special key,
+                    // but at least we shouldn't expose it to users.
+                    if (!Object.prototype.hasOwnProperty.call(command, SPECIAL_KEY)) {
+                        Object.defineProperty(command, SPECIAL_KEY, {
+                            configurable: true,
+                            enumerable: false,
+                            value: {},
+                        });
+                    }
+
+                    // We store the values inside the special
+                    // key, so that we can clean them up later.
+
+                    // @ts-expect-error: We've just defined this property
+                    command[SPECIAL_KEY][propertyName] = values;
+
+                    // We define a getter and setter combination to
+                    // silently push all assigned values into our store.
+                    Object.defineProperty(command, propertyName, {
+                        configurable: true,
+                        enumerable: true,
+
+                        // Because of the accessors, option transformers
+                        // should only set properties on the command via
+                        // regular assignment, only once per transformer.
+                        get: () => values[values.length - 1],
+                        set: (value) => values.push(value),
+                    });
+                });
+
+                this.registerCleanup(prototype, (command) => {
```

Comment:
I feel like this might be *too* generic, as it's not clear we'd really use it elsewhere 🤔 Instead of this `defineProperty` dance, what about keeping in the meta object a map of the action to take when assigning things into specific properties (a missing entry would just default to setting the value as usual), that the `Tuple` and `String` transformers would retrieve?

Doing this would also allow to put the Array decorator before or after the others, which might be a bit less confusing for users.